Firestore/Example/Tests/Local/FSTMutationQueueTests.mm

```diff
@@ -434,15 +438,15 @@ - (NSUInteger)batchCount {
  * @param batches The array to mutate, removing entries from it.
  * @return A new array containing all the entries that were removed from @a batches.
  */
-- (NSArray<FSTMutationBatch *> *)removeFirstBatches:(NSUInteger)n
-                                          inBatches:(NSMutableArray<FSTMutationBatch *> *)batches {
-  NSArray<FSTMutationBatch *> *removed = [batches subarrayWithRange:NSMakeRange(0, n)];
-  [batches removeObjectsInRange:NSMakeRange(0, n)];
-
-  [removed enumerateObjectsUsingBlock:^(FSTMutationBatch *batch, NSUInteger idx, BOOL *stop) {
-    [self.mutationQueue removeMutationBatch:batch];
-  }];
-
+- (std::vector<FSTMutationBatch *>)removeFirstBatches:(NSUInteger)n
+                                            inBatches:(std::vector<FSTMutationBatch *> *)batches {
+  std::vector<FSTMutationBatch *> removed;
+  for (int i = 0; i < n; i++) {
+    FSTMutationBatch *batch = *batches->begin();
+    removed.push_back(batch);
+    batches->erase(batches->begin());
+    self.mutationQueue->RemoveMutationBatch(batch);
+  }
```

Comment:
Alternative:

```
vector<FSTMutationBatch*> removed(batches.begin(), batches.begin()+n);
batches.erase(batches.begin(), batches.begin()+n);

for (FSTMutationBatch* batch : removed) {
  self.mutationQueue->RemoveMutationBatch(batch);
}

return removed;
```

Reply:
Done, with the exception that `batches` remains a pointer to a vector rather than a reference. My understanding is that mutable references are frowned upon and `batches` needs to be modified by the function.